src/python/pants/jvm/resolve/lockfile_metadata.py

```diff
@@ -110,13 +101,14 @@ def _from_json_dict(
 
         return JVMLockfileMetadataV1(requirements)
 
-    def _header_dict(self) -> dict[Any, Any]:
-        out = super()._header_dict()
-
-        out["generated_with_requirements"] = (
-            sorted(self.requirements) if self.requirements is not None else None
-        )
-        return out
+    @classmethod
+    def header_attrs(cls, instance: LockfileMetadata) -> dict[Any, Any]:
+        instance = cast(JVMLockfileMetadataV1, instance)
```

Comment:
Does this work?

```suggestion
    def header_attrs(cls, instance: JVMLockfileMetadataV1) -> dict[Any, Any]:
```

I had success with type-tightening like this yesterday.

Reply:
Oh cool. Let me try that.

Reply:
No, it does not work:

```
src/python/pants/jvm/resolve/lockfile_metadata.py: note: In class "JVMLockfileMetadataV1":
src/python/pants/jvm/resolve/lockfile_metadata.py:102:5: error: Argument 1 of
"header_attrs" is incompatible with supertype "LockfileMetadata"; supertype
defines the argument type as "LockfileMetadata"  [override]
        def header_attrs(cls, instance: JVMLockfileMetadataV1) -> dict[Any...
        ^
src/python/pants/jvm/resolve/lockfile_metadata.py:102:5: note: This violates the Liskov substitution principle
src/python/pants/jvm/resolve/lockfile_metadata.py:102:5: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
Found 1 error in 1 file (checked 3 source files)
```